great_expectations/datasource/fluent/interfaces.py

```diff
@@ -488,6 +510,19 @@ def _execution_engine_type(self) -> Type[_ExecutionEngineT]:
         """Returns the execution engine to be used"""
         return self.execution_engine_override or self.execution_engine_type
 
+    def add_batch_config(self, batch_config: BatchConfig) -> BatchConfig:
+        if not self.data_context:
+            raise DataContextError("Datasource is not attached to a DataContext")
+        return self.data_context.datasource_store.add_batch_config(batch_config)
+
+    def is_persisted(self) -> bool:
+        if self._data_context:
+            store = self._data_context._datasource_store
```

Comment:
It would be preferable if the datasouce knew it's store and didn't have to reach into the context. That might mean hooking it up at instantiation time like we do to associated the asset with the batch config. However, this work can be done in a subsequent PR.

Reply:
Strong agree. Added https://greatexpectations.atlassian.net/browse/V1-115 to track this.